LibRZ/include/Surfaces/Conic.h

```diff
@@ -74,6 +75,167 @@ namespace RZ {
+        
+        Real r1 = sqrt(y1);
+        Real r2 = sqrt(y2);
+        
+        Real eq1 = Ap + Bp * sqrt(1 + Cp * r1 * r1) - An - Bn * sqrt(1 + Cn * r1 * r1) + t;
+        Real eq2 = Ap + Bp * sqrt(1 + Cp * r2 * r2) - An - Bn * sqrt(1 + Cn * r2 * r2) + t;
+        
+        if (abs(eq1) < eps){
+          Rmax = sqrt(y1);
+        } else if (abs(eq2) < eps) {
+          Rmax = sqrt(y2);
+        } else {
+          RZWarning("There are no real solutions.\n");
+          Rmax = NAN;
+        }
+      } else {
+        Real y = -c / b;
+        if (y < 0)
+        {
+          RZWarning("There are no real solutions.\n");
+          Rmax = NAN;
+        }
+        Rmax = sqrt(y);
+      }
+
+      return Rmax;
+    }
+    
+    static inline Real
+    RmaxParNonPar(Real Sp, Real Rcp, Real Kp, Real dp, Real Sn, Real Rcn, Real Kn, Real dn, Real t)
+    {
+      Real Rmax = 0;
+      Real Rc2p = Rcp * Rcp;
+      Real Rc2n = Rcn * Rcn;
+
+      Real A = Sp * (dp * (Kp + 1) - Rcp) / (Kp + 1);
+      Real B = Sp * Rcp / (Kp + 1);
+      Real C = - (Kp + 1) / Rc2p;
+
+      Real X = Sn * dn;
+      Real Y = -Sn / (2 * Rcn);
+
+      Real D = X - A - t;
+        
+      Real B2 = B * B;
+      Real Y2 = Y * Y;
+      Real D2 = D * D;
+
+      Real a = Y2;
+      Real b = 2 * Y * D - B2 * C;
+      Real c = D2 - B2;
+        
+      Real eps = 1e-12;
+      if (abs(a) > eps) {
+        Real disc = b * b - 4 * a * c;
+        if (disc < 0) {
+          RZWarning("There are no real solutions.\n");
+          Rmax = NAN;
+        }
+        Real y1 = (-b + sqrt(disc)) / (2 * a);
+        Real y2 = (-b - sqrt(disc)) / (2 * a);
+        
+        Real r1 = sqrt(y1);
+        Real r2 = sqrt(y2);
+        
+        Real eq1 = X + Y * r1 * r1 - A - B * sqrt(1 + C * r1 * r1) + t;
+        Real eq2 = X + Y * r2 * r2 - A - B * sqrt(1 + C * r2 * r2) + t;
+        
+        if (abs(eq1) < eps){
+          Rmax = sqrt(y1);
+        } else if (abs(eq2) < eps) {
+          Rmax = sqrt(y2);
+        } else {
+          RZWarning("There are no real solutions.\n");
+          Rmax = NAN;
+        }
+      } else {
+        Real y = -c / b;
+        if (y < 0) {
+          RZWarning("There are no real solutions.\n");
+          Rmax = NAN;
+        } 
+        Rmax = sqrt(y);
+      }
+      
+      return Rmax;
+    }
+    
+    static inline Real
+    RmaxParPar(Real Sp, Real Rcp, Real Kp, Real dp, Real Sn, Real Rcn, Real Kn, Real dn, Real t)
+    {
+      Real Rmax = 0;
+      Real Xp = Sp * dp;
+      Real Yp = -Sp / (2 * Rcp);
+      Real Xn = Sn * dn;
+      Real Yn = -Sn / (2 * Rcn);
+        
+      Rmax = sqrt((Xn - Xp - t) / (Yp - Yn));
+      
+      return Rmax;
+    }
+
+    static inline Real
+    Rmax(Real Sp, Real Rcp, Real Kp, Real dp, Real Sn, Real Rcn, Real Kn, Real dn, Real t) {
+      Real Rmax;
+      bool isParabolicP = isZero(Kp + 1);
+      bool isParabolicN = isZero(Kn + 1);
+      if (!isParabolicP && !isParabolicN) {
```

Comment:
When you have several `if` / `else if` / `else` statements with only one sentence per block, you should remove the braces altogether:

```c
if (cond1)
  Rmax = doSomething1();
else if (cond2)
  Rmax = doSomething2();
else if (cond3)
  Rmax = doSomething3();
else
  Rmax = doSomethingElse();  
```